securedrop_client/logic.py

```diff
@@ -459,7 +459,7 @@ def on_sync_failure(self, result: Exception) -> None:
         a sync fails is ApiInaccessibleError then we need to log the user out for security reasons
         and show them the login window in order to get a new token.
         """
-        logger.debug('The SecureDrop server cannot be reached due to Error: {}'.format(result))
+        logger.error('sync failure: %s', result)
```

Comment:
nit: i personally prefer using `.format` over `%s` because it protects from type errors in certain circumstances and is easier to read imo

Reply:
Making logging calls with % formatting and positional arguments defers [interpolation](https://github.com/python/cpython/blob/d730719b094cb006711b1cd546927b863c173b31/Lib/logging/__init__.py#L371) until the message is actually going to be logged. It doesn't matter at all here, since I'm promoting the call to `error` from `debug`, so it is always going to happen, and removing the overhead of string interpolation in log messages is probably not going to change the client's performance characteristics. But every little bit helps, and I think this approach is a good habit in general.

If we want to choose `.format` or f-strings as a project style policy, we might also consider wider use of [isEnabledFor](https://docs.python.org/3/howto/logging.html#optimization) before logging calls.

Reply:
I see you want the small performance gain by using `%s`... I thought that was going to get deprecated at some point, I guess not. I'd like to look at the performance gains of using the old style at some point, but I don't feel too strongly about you introducing it into the client code, although it goes against the implicit style of string formatting since it's not used anywhere else in the code base.

Reply:
I'm all about the small performance gains, but I repented of my maverick ways and now it's back to `.format`.